diesel/src/expression/array_comparison.rs

```diff
@@ -73,13 +73,17 @@ impl<T, U> NonAggregate for NotIn<T, U> where
 impl<T, U, DB> QueryFragment<DB> for In<T, U> where
     DB: Backend,
     T: QueryFragment<DB>,
-    U: QueryFragment<DB>,
+    U: QueryFragment<DB> + MaybeEmpty,
 {
     fn to_sql(&self, out: &mut DB::QueryBuilder) -> BuildQueryResult {
-        try!(self.left.to_sql(out));
-        out.push_sql(" IN (");
-        try!(self.values.to_sql(out));
-        out.push_sql(")");
+        if self.values.is_empty() {
+            out.push_sql("1=0");
```

Comment:
Maybe add a comment to say this is just to return a valid sql expression that always evals to false?